core/shared/src/main/scala/zio/prelude/NonEmptyList.scala

```diff
@@ -78,6 +78,21 @@ sealed trait NonEmptyList[+A] { self =>
       if (seen(a)) (as, seen) else (cons(a, as), seen + a)
     }._1.reverse
 
+  /** Decomposes the `NonEmptyList` into an element and a (possibly empty) `List` */
+  final def peel: (A, List[A]) = self match {
+    case Single(head)     => (head, List())
+    case Cons(head, tail) => (head, tail.toList)
+  }
+
+  /**
+   * Decomposes the `NonEmptyList` either into an element and the remaining `NonEmptyList`,
+   * or just a single element, if there aren't any other.
+   */
+  final def peelEither: Either[A, (A, NonEmptyList[A])] = self match {
```

Comment:
Should this be an `Either` or an `(A, Option[NonEmptyList[A])`. They are isomorphic but often it makes sense to factor out the common term of the sum type so it is really reflecting what is different. You always have an `A` and then you may or may not have another `NonEmptyList[A]`.

Reply:
Good idea :+1: